pkg/discovery/pods.go

```diff
@@ -63,19 +63,25 @@ func getPodLogOptions(cfg *config.Config) *v1.PodLogOptions {
 }
 
 // gatherPodLogs will loop through collecting pod logs and placing them into a directory tree
-func gatherPodLogs(kubeClient kubernetes.Interface, ns string, opts metav1.ListOptions, cfg *config.Config) error {
+func gatherPodLogs(kubeClient kubernetes.Interface, ns string, opts metav1.ListOptions, cfg *config.Config,
```

Comment:
I'd modify the comment to explain why you're adding the `visitedPods` field (e.g. that this may be run in a namespace with varying listoptions which might overlap).

tiny nit: I'd modify it to be `map[string]struct{}` :
 - makes it clear that we don't really want a map of `true|false` values but that we are just using the map as a list of names which can be checked quickly
 - saves a tiny bit of space (not that there is a memory concern here really but just FYI)